src/assets/wise5/authoringTool/addLesson/add-lesson-configure/add-lesson-configure.component.ts

```diff
@@ -11,18 +13,46 @@ export class AddLessonConfigureComponent {
   addLessonFormGroup: FormGroup = this.fb.group({
     title: new FormControl('', [Validators.required])
   });
+  target: string;
   @ViewChild('titleField') titleField: ElementRef;
 
-  constructor(private fb: FormBuilder, private route: ActivatedRoute, private router: Router) {}
+  constructor(
+    private fb: FormBuilder,
+    private route: ActivatedRoute,
+    private projectService: TeacherProjectService,
+    private router: Router
+  ) {}
 
-  ngAfterViewInit() {
+  ngOnInit(): void {
+    this.target = history.state.target;
+  }
+
+  ngAfterViewInit(): void {
     this.titleField.nativeElement.focus();
   }
 
-  protected next(): void {
-    this.router.navigate(['../choose-location'], {
-      relativeTo: this.route,
-      state: { title: this.addLessonFormGroup.controls['title'].value }
+  protected submit(): void {
+    const newLesson = this.createNewLesson();
```

Comment:
Remove createNewLesson() function since it's just one line and is only called from here?
```suggestion
    const newLesson = this.projectService.createGroup(this.addLessonFormGroup.controls['title'].value);
```